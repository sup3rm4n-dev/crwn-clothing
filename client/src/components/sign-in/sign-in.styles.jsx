import styled from 'styled-components';

/* CSS */
export const SignInContainer = styled.div`
  width: 380px;
  display: flex;
  flex-direction: column;
`;

export const SignInTitle = styled.h2`
  margin: 10px 0;
`;

export const ButtonsBarContainer = styled.div`
  display: flex;
  justify-content: space-between;
  box-sizing: border-box;
`;
