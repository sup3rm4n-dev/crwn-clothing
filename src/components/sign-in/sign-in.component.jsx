import React, { Component } from 'react';
import { auth, signInWithGoogle } from '../../firebase/firebase.utils';
import CustomButton from '../custom-button/custom-button.component';
import FormInput from '../form-input/form-input.component';
import {
  ButtonsBarContainer,
  SignInContainer,
  SignInTitle,
} from './sign-in.styles';

class SignIn extends Component {
  state = {
    email: '',
    password: '',
  };

  handleSubmit = async event => {
    event.preventDefault();

    const { email, password } = this.state;

    try {
      await auth.signInWithEmailAndPassword(email, password);
      this.setState({ email: '', password: '' });
    } catch (error) {
      console.log(error);
    }
  };

  handleChange = event => {
    const { value, name } = event.target;

    this.setState({ [name]: value });
  };

  render() {
    return (
      <SignInContainer>
        <SignInTitle>I already have an account</SignInTitle>
        <span>Sign in with your email password</span>

        <form onSubmit={this.handleSubmit}>
          <FormInput
            name='email'
            type='email'
            value={this.state.email}
            handleChange={this.handleChange}
            label='email'
            required
            autoComplete='username'
          />
          <FormInput
            name='password'
            type='password'
            value={this.state.password}
            handleChange={this.handleChange}
            label='password'
            required
            autoComplete='current-password'
          />
          <ButtonsBarContainer>
            <CustomButton type='submit'>Sign In</CustomButton>
            <CustomButton onClick={signInWithGoogle} isGoogleSignIn>
              Sign in with Google
            </CustomButton>
          </ButtonsBarContainer>
        </form>
      </SignInContainer>
    );
  }
}

export default SignIn;
