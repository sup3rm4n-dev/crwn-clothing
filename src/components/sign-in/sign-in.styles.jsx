import styled from 'styled-components/macro';

/* CSS */
export const SignInContainer = styled.div`
  width: 380px;
  display: flex;
  flex-direction: column;

  @media screen and (max-width: 400px) {
    margin: 10px;
    width: 300px;
  }
`;

export const SignInTitle = styled.h2`
  margin: 10px 0;
`;

export const ButtonsBarContainer = styled.div`
  display: flex;
  justify-content: space-between;
  box-sizing: border-box;

  @media screen and (max-width: 400px) {
    margin: 10px;
    width: 300px;
  }
`;
