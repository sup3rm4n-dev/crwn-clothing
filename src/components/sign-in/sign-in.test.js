import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from './sign-in.component';

describe('SignIn component', () => {
  let wrapper;
  let mockEmailSignInStart;
  let mockGoogleSignInStart;

  beforeEach(() => {
    mockEmailSignInStart = jest.fn();
    mockGoogleSignInStart = jest.fn();

    const mockProps = { mockEmailSignInStart, mockGoogleSignInStart };
    wrapper = shallow(<SignIn {...mockProps} />);
  });

  it('should render SignIn Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an email and password input elements with value and handleChange props', () => {
    const emailInput = wrapper.find('FormInput').at(0);
    const passwordInput = wrapper.find('FormInput').at(1);
    expect(emailInput.props()).toHaveProperty('value', '');
    expect(passwordInput.props()).toHaveProperty('value', '');
    expect(emailInput.props()).toHaveProperty(
      'handleChange',
      expect.any(Function)
    );
    expect(passwordInput.props()).toHaveProperty(
      'handleChange',
      expect.any(Function)
    );
  });

  it('should have onSubmit prop attached to form element', () => {
    expect(wrapper.find('#form').props()).toHaveProperty(
      'onSubmit',
      expect.any(Function)
    );
  });
});
