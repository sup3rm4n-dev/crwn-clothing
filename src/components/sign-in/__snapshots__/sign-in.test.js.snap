// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn Component 1`] = `"<div class=\\"SignInContainer-sc-11fykje-0 cLbcfm\\"><h2 class=\\"SignInTitle-sc-11fykje-1 cYmKHy\\">I already have an account</h2><span>Sign in with your email and password</span><form id=\\"form\\"><div class=\\"GroupContainer-sc-96a6za-0 hoKTZU\\"><input type=\\"email\\" data-testid=\\"email-input\\" name=\\"email\\" value=\\"\\" aria-label=\\"Email\\" autoComplete=\\"username\\" aria-required=\\"true\\" required=\\"\\" class=\\"FormInputContainer-sc-96a6za-1 fpriDh\\"/><label class=\\"FormInputLabel-sc-96a6za-2 HJNXi\\">Email</label></div><div class=\\"GroupContainer-sc-96a6za-0 hoKTZU\\"><input type=\\"password\\" data-testid=\\"password-input\\" name=\\"password\\" value=\\"\\" aria-label=\\"Password\\" autoComplete=\\"current-password\\" aria-required=\\"true\\" required=\\"\\" class=\\"FormInputContainer-sc-96a6za-1 fpriDh\\"/><label class=\\"FormInputLabel-sc-96a6za-2 HJNXi\\">Password</label></div><div class=\\"ButtonsBarContainer-sc-11fykje-2 cOMDpH\\"><button type=\\"submit\\" class=\\"CustomButtonContainer-ttlcs0-0 YaHjW\\"> Sign in </button><button type=\\"button\\" class=\\"CustomButtonContainer-ttlcs0-0 eJidjN\\">Sign in with Google</button></div></form></div>"`;
